### PR TITLE
Data warehouse changes

### PR DESCRIPTION
**Pull request by meyerovb**

### CODE REVIEW

File: lib/purview/databases/base.rb [hunk 1]
```diff
@@ -446,6 +466,11 @@ def next_window(connection, table, timestamp)
         )
         max = min + table.window_size
```
Comment (mattdoller) [inline lib/purview/databases/base.rb]: can the setting of these mins and maxs be put into separate methods?  it's incredibly hard to follow the logic through this method - max is set to min + table.window_size, then it's checked against the highest min and possibly set to min + baseline_window, then inside the Window instantiation, it's checked again against now and set to either now or max.  can this be cleaned up a bit?

File: lib/purview/types.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 require 'purview/types/date'
 require 'purview/types/float'
 require 'purview/types/integer'
+require 'purview/types/bigint'
```
Comment (mattdoller) [inline lib/purview/types.rb:7]: isn't there a typemap that also needs to be updated which maps a purview column type to a sql column type name?

Comment (meyerovb) [inline lib/purview/types.rb:7]: Already done https://github.com/sermo/purview/commit/94a443666f479b8add0563eca5c0a644b1f2430f#diff-48e3950e2aee372647981dc71d803828R567

I did add a different bit I missed though

File: lib/purview/databases/base.rb (hunk 1 above)
Comment (meyerovb) [inline lib/purview/databases/base.rb]: Cleaned up
